main/lp/lp_add_author.php

```diff
@@ -0,0 +1,379 @@
+if ($idExtraField != 0) {
+    $extraFieldValueUser = new ExtraFieldValue('user');
+    $arrayExtraFieldValueUser = $extraFieldValueUser->get_item_id_from_field_variable_and_field_value(
+        $authorLp['variable'],
+        1,
+        true,
+        false,
+        true);
+
+    foreach ($arrayExtraFieldValueUser as $item) {
+        $teacher = api_get_user_info($item['item_id']);
+        $teachers[] = $teacher;
+    }
+}
+/* Authors*/
+foreach ($teachers as $key => $value) {
+    $authorId = $value['id'];
+    $authorName = $value['complete_name'];
+    if (!empty($authorName)) {
+        $options[$authorId] = $authorName;
+        if (isset($defaultAuthor[$authorId])) {
+            // $default["authorItemSelect"][$authorId] = $authorId;
+        }
+    }
+}
+$form->addSelect('authorItemSelect', get_lang('Authors'), $options, [
+    'multiple' => 'multiple',
+]);
+$form->addHtml('</div>');
+$form->addButtonCreate(get_lang('Send'));
+$form->setDefaults($default);
+$form->display();
+echo '</div>';
+echo '</div>';
+if ($form->validate()) {
+    if (isset($_GET['sub_action']) && ($_GET['sub_action'] === 'author_view')) {
+        $authors = $_POST['authorItemSelect'];
+        $items = $_POST['itemSelected'];
+        unset($author);
+        $saveExtraFieldItem = [];
+        $saveAuthor = [];
+        $removeExist = 0;
+        foreach ($_SESSION['oLP']->items as $item) {
+            $itemName = $item->name;
+            $itemId = $item->iId;
+            if (isset($items[$itemId])) {
+                if (count($authors) == 0) {
+                    $saveExtraFieldItem[$itemId][0] = 0;
+                }
+                foreach ($authors as $author) {
+                    if ($author == 0 || $removeExist == 1) {
+                        $saveExtraFieldItem[$itemId][0] = 0;
+                        $removeExist = 1;
+                    } else {
+                        $saveExtraFieldItem[$itemId][$author] = $author;
+                    }
+                }
+            }
+        }
+
+        if (count($saveExtraFieldItem) > 0) {
+            $messages = '';
+            $lastEdited = [];
+            foreach ($saveExtraFieldItem as $saveItemId => $values) {
+                $extraFieldValues = $extraFieldValue->get_values_by_handler_and_field_variable(
+                    $saveItemId,
+                    'AuthorLPItem'
+                );
+                $extraFieldValue->save([
+                    'variable' => 'AuthorLPItem',
+                    'value' => $values,
+                    'item_id' => $saveItemId,
+                ]);
+                $lastEdited = $values;
+                $saveAuthor[] = $options[$author];
+            }
+            $saveAuthor = array_unique($saveAuthor);
+            $messages .= implode(' / ', $saveAuthor);
+            $currentUrl = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http")."://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]";
```

Comment:
Avoid using unquoted array indexes in a string.

Reply:
We prefer using api_get_self(), que usa `$_SERVER['PHP_SELF']` para expresar lo mismo (creo): https://www.php.net/manual/en/reserved.variables.server.php

Reply:
With api_get_self() it would not work, since the query of the GET method is required to be able to access the author edition. However api_request_uri()is possible since it calls directly $ _SERVER ['REQUEST_URI']